crates/astria-sequencer-relayer/src/relayer.rs

```diff
@@ -93,8 +92,20 @@ impl Relayer {
         let sequencer = HttpClient::new(&*cfg.sequencer_endpoint)
             .wrap_err("failed to create sequencer client")?;
 
-        let validator = Validator::from_path(&cfg.validator_key_file)
-            .wrap_err("failed to get validator info from file")?;
+        let validator = if cfg.disable_relay_all {
```

Comment:
This can be expressed a bit neater without right-drift if-let chains.
```rust
let validator = match (cfg.disable_relay_all, cfg.validator_key_file) {
    (true, Some(file)) => Some(validator::from_path(file).wrap_err("baaaaad")?),
    (true, None) => bail!("disable relay set, but no file, etc etc"),
    (false, _) => None, // could also say that the file was unnecessarily set, but it's ok
};

Reply:
oh cool :D